tests/unit/kg_construction/test_schema.py

```diff
@@ -0,0 +1,234 @@
+#  Copyright (c) "Neo4j"
+#  Neo4j Sweden AB [https://neo4j.com]
+#  #
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  #
+#      https://www.apache.org/licenses/LICENSE-2.0
+#  #
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+from __future__ import annotations
+
+import pytest
+from neo4j_genai.exceptions import SchemaValidationError
+from neo4j_genai.kg_construction.schema import Entity, Relation, SchemaBuilder
+from pydantic import ValidationError
+
+
+# Fixtures for test data
+@pytest.fixture
+def valid_entities() -> list[Entity]:
+    return [
+        Entity(name="PERSON", type="str", description="An individual human being."),
```

Comment:
I think only now I really understand this schema format :D It seems to be really bounds to RDF, where entities are flat (no properties).

Reply:
I see, do you see a problem with this approach? Perhaps we can add "properties" to Entity and Relation ?

Reply:
https://github.com/neo4j/neo4j-genai-python/pull/88/commits/e7be9357a3f6a447d9d00e654abb569465f7755b added here

Reply:
Hey Will, sorry I've been thinking about the question quite a while, basically my concern was whether we should support ontologies AND labelled property graph (LPG) in the same object, and my conclusion is that it will be hard. WDYT?
I tend to think we should focus on LPG right now, so yes adding properties to entities and assuming the entities we extract are (for most of them) the final nodes we will store.
If we agree on this approach, I think properties should be objects, because it is useful to include a type (GPT-4o is able to normalize dates for instance, which is super useful), and a description could also help in some circumstances (but this last assertion is untested)